docs/migration-guide-v2.md

```diff
@@ -0,0 +1,214 @@
+# Parser v1 to v2 Migration Guide
```

Comment:
Maybe it makes sense to start of with "why v2" 🤔

Reply:
Overall, the first point of the General indicates why v2 - because a lot of breaking changes. Do we really need some description for that?

Reply:
I guess not no 😄